Source/Core/VideoCommon/PerformanceTracker.h

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <array>
+#include <chrono>
+#include <fstream>
+#include <mutex>
+#include <optional>
+
+#include "Common/CommonTypes.h"
+
+class PerformanceTracker
+{
+private:
+  // Must be powers of 2 for masking to work
+  static constexpr u64 MAX_DT_QUEUE_SIZE = 1UL << 12;
+  static constexpr u64 MAX_QUALITY_GRAPH_SIZE = 1UL << 8;
+
+  static inline std::size_t IncrementIndex(const std::size_t index)
```

Comment:
```suggestion
  static std::size_t IncrementIndex(const std::size_t index)
```

member and static member functions implemented in the definition are inline by default